src/mmw/js/src/analyze/views.js

```diff
@@ -238,60 +202,71 @@ var TabContentView = Marionette.LayoutView.extend({
         role: 'tabpanel'
     },
     regions: {
-        aoiRegion: '.analyze-aoi-region',
-        tableRegion: '.analyze-table-region',
-        chartRegion: '.analyze-chart-region'
+        aoiRegion: '.aoi-region',
+        resultRegion: '.result-region'
     },
-    onShow: function() {
-        var categories = this.model.get('categories'),
-            largestArea = _.max(_.pluck(categories, 'area')),
-            units = utils.magnitudeOfArea(largestArea);
-
-        var census;
-        switch (this.model.get('name')) {
-            case ('land'):
-                census = new coreModels.LandUseCensusCollection(categories);
-                break;
-            case ('soil'):
-                census = new coreModels.SoilCensusCollection(categories);
-                break;
-            case ('animals'):
-                census = new coreModels.AnimalCensusCollection(categories);
-                break;
-            case ('pointsource'):
-                census = new coreModels.PointSourceCensusCollection(categories);
-                break;
-            default:
-                throw new Error('invalid CensusCollection');
-        }
 
+    onShow: function() {
         this.aoiRegion.show(new AoiView({
             model: new coreModels.GeoModel({
                 place: App.map.get('areaOfInterestName'),
                 shape: App.map.get('areaOfInterest')
             })
         }));
 
-        if (this.model.get('name') === 'animals') {
-            this.tableRegion.show(new AnimalTableView({
-                units: units,
-                collection: census
-            }));
-        } else if (this.model.get('name') === 'pointsource') {
-            this.tableRegion.show(new PointSourceTableView({
-                units: units,
-                collection: census
-            }));
+        this.showAnalyzingMessage();
+
+        this.model.get('taskRunner').fetchAnalysisIfNeeded()
+            .done(_.bind(this.showResultsIfNotDestroyed, this))
+            .fail(_.bind(this.showErrorIfNotDestroyed, this));
```

Comment:
We need the `IfNotDestroyed` versions for the case when loading an existing project, and the Analyze requests are fired but the Analyze view is not shown. Otherwise, it tries to show views in regions that don't exist and blows up.